File: java-api/src/main/java/de/davelee/personalman/api/UpdateSalaryRequest.java
package de.davelee.personalman.api;

import lombok.*;

/**
 * This class is part of the PersonalMan REST API. It represents a request for a registration of a new user.
 * @author Dave Lee
 */
@Builder
@Getter
@AllArgsConstructor
@NoArgsConstructor
@Setter
public class UpdateSalaryRequest {

    //company that the person works for
    private String company;

    //username of the person
    private String username;

    //token of the user making the change
    private String token;

    //hourly wage that the person should get
    private Double hourlyWage;

    //number of hour person works per week
    private int contractedHoursPerWeek;

}

File: server/src/users/requests/updatesalary.request.ts
import {ApiProperty} from "@nestjs/swagger";

export class UpdateSalaryRequest {

    //company that the person works for
    @ApiProperty()
    private company: string;

    //username of the person
    @ApiProperty()
    private username: string;

    //token of the user making the change
    @ApiProperty()
    private token: string;

    //hourly wage that the person should get
    @ApiProperty()
    private hourlyWage: number;

    //number of hour person works per week
    @ApiProperty()
    private contractedHoursPerWeek: number;

    getCompany(): string {
        return this.company;
    }

    getUsername(): string {
        return this.username;
    }

    getToken(): string {
        return this.token;
    }

    getHourlyWage(): number {
        return this.hourlyWage;
    }

    getContractedHoursPerWeek(): number {
        return this.contractedHoursPerWeek;
    }

}